### ISSUE TITLE
Drop support for axum-jsonschema

### ISSUE THREAD

**Issue by tamasfe** [tamasfe/aide#162]
It's a tiny library that is just an opinionated glue of other libraries that feels a bit like `is-odd`.

In its current state it didn't fit my original use-case anymore and I didn't think it would be worth the hassle to make it more flexible/configurable since I could copy the code and modify it to my needs.

I don't think it is worth the maintenance burden of juggling different upstream library versions, maybe I could move it to its own repo or gist as a reference for others.

**Comment by tamasfe** [comment]
Since only I have access to the crates.io package currently, I'll just move it to its own repo and archive it if everyone agrees.

**Comment by jplatte** [comment]
Right, that makes sense. In that case the axum-extra stuff should maybe be separated out as well? It's a mess really...

Personally I wouldn't mind more crates though. I have been maintaining larger workspaces, the important part is just to automate the release process a bit.

Could even do `aide-core` for the traits, `aide-axum` for the axum integration, and so on..

**Comment by tamasfe** [comment]
> Could even do `aide-core` for the traits, `aide-axum` for the axum integration, and so on..

You'd need to wrap all the axum types due to orphan rules. I didn't want to do this so that aide is as much as a drop-in as possible. I'm not sure about user experience, but I think as it stands currently you can get documentation for your existing axum application by replacing 2-3 imports, we'd lose this if we went the wrapper-way (we'd also have to keep up with every upstream change even if they don't necessarily concern aide (e.g. new methods on extractors)).

I think the least messy way would be to just go the feature-gated way supporting a specific version set at a time (e.g. latest compatible at the point of the release), if anyone diverges from that, they'll have to create their own wrappers.

**Comment by tamasfe** [comment]
> Right.. So those two extra crates use wrapper types then?

I assume, yes, there is no other way for them to be in separate crates afaik.

**Comment by jplatte** [comment]
Dropped all the auxiliary crates in #171 for now. I can't publish them anyways, and it shouldn't be a huge effort to re-introduce more things again should people ask for it.
